admob/src/android/admob_android.cc

```diff
@@ -980,39 +876,90 @@ Java_com_google_firebase_admob_internal_cpp_InterstitialAdHelper_notifyAdFailedT
   // Invoke AdMobInternal, a friend of AdResult, to have it access its
   // protected constructor with the AdError data.
   const AdResult& ad_result = AdMobInternal::CreateAdResult(ad_result_internal);
-  internal->NotifyListenerOfAdFailedToShowFullScreenContent(ad_result);
+  listener->NotifyListenerOfAdFailedToShowFullScreenContent(ad_result);
 }
 
-extern "C" JNIEXPORT void JNICALL
-Java_com_google_firebase_admob_internal_cpp_InterstitialAdHelper_notifyAdImpressionEvent(  // NOLINT
+extern "C" JNIEXPORT void JNICALL JNI_notifyAdImpressionEvent(JNIEnv* env,
```

Comment:
I believe you don't need any of the "extern "C" JNIEXPORT JNICALL stuff since we are registering natives directly - also it would probably be best to put these in an unnamed namespace. (along with the one I added!)

Reply:
Done.